File: Demo/Objective_C_Demo/ViewController/YYTextViewController.m
//
//  YYTextViewController.m
//  Demo
//
//  Created by IEMacBook01 on 21/05/16.
//  Copyright © 2016 Iftekhar. All rights reserved.
//

#import "YYTextViewController.h"

@import IQKeyboardManager;
@import YYText;

@interface YYTextViewController ()<YYTextViewDelegate>

@end

@implementation YYTextViewController
{
    IBOutlet YYTextView *textView;
}

-(void)dealloc
{
    textView = nil;
}

+(void)initialize
{
    [super initialize];
    
    [[IQKeyboardManager sharedManager] registerTextFieldViewClass:[YYTextView class] didBeginEditingNotificationName:YYTextViewTextDidBeginEditingNotification didEndEditingNotificationName:YYTextViewTextDidEndEditingNotification];
}

-(void)viewDidLoad
{
    [super viewDidLoad];
    textView.placeholderText=@"This is placeholder text of YYTextView";
}

- (void)textViewDidBeginEditing:(YYTextView *)tv
{
    [tv reloadInputViews];
}

@end

File: Demo/Swift_Demo/ViewController/YYTextViewController.swift
//
//  YYTextViewController.swift
//  Demo
//
//  Created by IEMacBook01 on 23/05/16.
//  Copyright © 2016 Iftekhar. All rights reserved.
//

import IQKeyboardManagerSwift
import YYText

class YYTextViewController: UIViewController, YYTextViewDelegate {

    @IBOutlet var textView : YYTextView!

    deinit {
        textView = nil
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        
        IQKeyboardManager.shared.registerTextFieldViewClass(YYTextView.self, didBeginEditingNotificationName: Notification.Name.YYTextViewTextDidBeginEditing.rawValue, didEndEditingNotificationName: Notification.Name.YYTextViewTextDidEndEditing.rawValue)

        textView.placeholderText = "This is placeholder text of YYTextView"
    }
    
    func textViewDidBeginEditing(_ tv: YYTextView) {
        tv.reloadInputViews()
    }
}
